src/main/java/com/minecolonies/colony/buildings/AbstractBuilding.java

```diff
@@ -503,6 +506,18 @@ public final void destroy()
      */
     public void onDestroyed()
     {
+
+        final TileEntityColonyBuilding tileEntity = this.getTileEntity();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "tileEntity" which hides the field declared at line 125. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)